tests/integration/template_min_case_e3sm_diags_lat_lon_land_mvm_2.cfg

```diff
@@ -34,7 +34,7 @@ short_name = "#expand case_name#"
 walltime = "#expand diags_walltime#"
 
   [[ lnd_monthly_mvm_lnd ]]
-  climo_subsection = "land_monthly_climo"
+  climo_land_subsection = "land_monthly_climo"
```

Comment:
Key Change 2: specifying a climo subsection specific to land

Reply:
@forsyth2 I have a clarification question regarding to backward compatibility. What would be the consequence if in an old zppy config file, a user uses  ` climo_subsection = "land_monthly_climo"`

Reply:
Ah, so that wouldn't have actually had any effect. And therefore there is nothing to be backwards compatible with.

This block is new in this PR:
```
    if "lat_lon_land" in c["sets"]:
        check_parameter_defined(c, "climo_land_subsection")
        dependencies.append(
            os.path.join(
                script_dir, f"climo_{c['climo_land_subsection']}{status_suffix}"
            )
        )
```
Notice, `lat_lon_land` wasn't included in:
```
def add_climo_dependencies(
    c: Dict[str, Any], dependencies: List[str], script_dir: str
) -> None:
    depend_on_climo: Set[str] = set(
        [
            "lat_lon",
            "zonal_mean_xy",
            "zonal_mean_2d",
            "polar",
            "cosp_histogram",
            "meridional_mean_2d",
            "annual_cycle_zonal_mean",
            "zonal_mean_2d_stratosphere",
            "aerosol_aeronet",
            "aerosol_budget",
        ]
    )
```
So basically, including `climo_subsection` as a parameter in this block of the cfg was simply unused information before this PR.

It only _appeared_ like it was useful, because the land climo task often just happens to finish before the `lnd_monthly_mvm_lnd` diags task runs anyway.